rslint_core/src/groups/errors/no_extra_boolean_cast.rs

```diff
@@ -0,0 +1,414 @@
+use crate::rule_prelude::*;
+use ast::*;
+use SyntaxKind::*;
+
+declare_lint! {
+    /**
+     * In contexts where expression will be coerced to a `Boolean` (e.g. `if`),
+     * casting to a boolean (using `!!` or `Boolean(expr)`) is unnecessary.
+     *
+     * # Invalid Code Examples
+     * ```ignore
+     * if (!!foo) {}
+     * while (!!foo) {}
+     *
+     * var foo = !!!bar;
+     * var foo = Boolean(!!bar);
+     * ```
+     */
+    #[derive(Default)]
+    #[serde(default)]
+    NoExtraBooleanCast,
+    errors,
+    "no-extra-boolean-cast",
+
+    /**
+     * If this option is `true`, this rule will also check for unnecessary boolean
+     * cast inside logical expression, which is disabled by default.
+     */
```

Comment:
I would rather this be kept as `///` comments since it is not very large and its consistent with other rules, and there should be no newline between the code and the comment